In [10]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [3]:
df1 = pd.read_csv('stats_one.csv')
df2 = pd.read_csv('stats_two.csv')
df3 = pd.read_csv('stats_three.csv')
df4 = pd.read_csv('stats_four.csv')
df5 = pd.read_csv('stats_five.csv')

In [11]:
df1.columns

Index(['iteration', ' fittest_fit', ' average_fit',
       ' fittest_mountain_fitness', ' average_mountain_fitness',
       ' fittest_mountain_touching_fitness',
       ' average_mountain_touching_fitness', ' fittest_size_fitness',
       ' average_size_fitness', ' max_links', ' average_links'],
      dtype='object')

In [12]:
df2.columns

Index(['iteration', ' fittest_fit', ' average_fit',
       ' fittest_mountain_fitness', ' average_mountain_fitness',
       ' fittest_mountain_touching_fitness',
       ' average_mountain_touching_fitness', ' fittest_size_fitness',
       ' average_size_fitness', ' max_links', ' average_links'],
      dtype='object')

In [13]:
df3.columns

Index(['iteration', ' fittest_fit', ' average_fit',
       ' fittest_mountain_fitness', ' average_mountain_fitness',
       ' fittest_mountain_touching_fitness',
       ' average_mountain_touching_fitness', ' fittest_size_fitness',
       ' average_size_fitness', ' max_links', ' average_links'],
      dtype='object')

In [14]:
df4.columns

Index(['iteration', ' fittest_fit', ' average_fit',
       ' fittest_mountain_fitness', ' average_mountain_fitness',
       ' fittest_mountain_touching_fitness',
       ' average_mountain_touching_fitness', ' fittest_size_fitness',
       ' average_size_fitness', ' max_links', ' average_links'],
      dtype='object')

In [15]:
df5.columns

Index(['iteration', ' fittest_fit', ' average_fit',
       ' fittest_mountain_fitness', ' average_mountain_fitness',
       ' fittest_mountain_touching_fitness',
       ' average_mountain_touching_fitness', ' fittest_size_fitness',
       ' average_size_fitness', ' fittest_distance', ' mean_distance',
       ' fittest_vertical_distance', ' average_vertical_distance',
       ' max_links', ' average_links'],
      dtype='object')

In [4]:
def create_combined_graph(fittest_metric, average_metric, ylabel, filename):
    plt.figure(figsize=(15, 10))
    
    # Plot fittest metrics
    plt.plot(df1['iteration'], df1[fittest_metric], label='Run 1 Fittest (200 iter)', linestyle='-')
    plt.plot(df2['iteration'], df2[fittest_metric], label='Run 2 Fittest (200 iter)', linestyle='-')
    plt.plot(df3['iteration'], df3[fittest_metric], label='Run 3 Fittest (200 iter)', linestyle='-')
    plt.plot(df4['iteration'], df4[fittest_metric], label='Run 4 Fittest (150 iter)', linestyle='-')
    plt.plot(df5['iteration'], df5[fittest_metric], label='Run 5 Fittest (250 iter)', linestyle='-')
    
    # Plot average metrics
    plt.plot(df1['iteration'], df1[average_metric], label='Run 1 Average', linestyle='--')
    plt.plot(df2['iteration'], df2[average_metric], label='Run 2 Average', linestyle='--')
    plt.plot(df3['iteration'], df3[average_metric], label='Run 3 Average', linestyle='--')
    plt.plot(df4['iteration'], df4[average_metric], label='Run 4 Average', linestyle='--')
    plt.plot(df5['iteration'], df5[average_metric], label='Run 5 Average', linestyle='--')
    
    plt.title(f'Fittest and Average {ylabel} Over Iterations')
    plt.xlabel('Iteration')
    plt.ylabel(ylabel)
    plt.legend()
    plt.grid(True)
    plt.tight_layout()
    plt.savefig(filename)
    plt.close()

In [6]:
def create_combined_graph_new(fittest_metric, average_metric, ylabel, filename):
    plt.figure(figsize=(15, 10))
    
    # Plot fittest metrics
    plt.plot(df5['iteration'], df5[fittest_metric], label='Run 5 Fittest (250 iter)', linestyle='-')
    
    # Plot average metrics
    plt.plot(df5['iteration'], df5[average_metric], label='Run 5 Average', linestyle='--')
    
    plt.title(f'Fittest and Average {ylabel} Over Iterations')
    plt.xlabel('Iteration')
    plt.ylabel(ylabel)
    plt.legend()
    plt.grid(True)
    plt.tight_layout()
    plt.savefig(filename)
    plt.close()

In [9]:
create_combined_graph(' fittest_fit', ' average_fit', 'Fitness', 'combined_fitness.png')
create_combined_graph(' fittest_mountain_fitness', ' average_mountain_fitness', 'Mountain Fitness', 'combined_mountain_fitness.png')
create_combined_graph(' fittest_mountain_touching_fitness', ' average_mountain_touching_fitness', 'Mountain Touching Fitness', 'combined_mountain_touching_fitness.png')
create_combined_graph(' fittest_size_fitness', ' average_size_fitness', 'Size Fitness', 'combined_size_fitness.png')
create_combined_graph_new(' fittest_distance', ' mean_distance', 'Distance', 'combined_distance.png')
create_combined_graph_new(' fittest_vertical_distance', ' average_vertical_distance', 'Vertical Distance', 'combined_vertical_distance.png')

print("Combined graphs have been created and saved.")

Combined graphs have been created and saved.


In [22]:
def create_summary_table(df, params):
    # Calculate summary statistics
    best_fitness = df[' fittest_fit'].max()
    avg_fitness = df[' average_fit'].mean()
    
    best_mountain_fitness = df[' fittest_mountain_fitness'].max()
    avg_mountain_fitness = df[' average_mountain_fitness'].mean()
    
    best_mountain_touching_fitness = df[' fittest_mountain_touching_fitness'].max()
    avg_mountain_touching_fitness = df[' average_mountain_touching_fitness'].mean()
    
    best_size_fitness = df[' fittest_size_fitness'].max()
    avg_size_fitness = df[' average_size_fitness'].mean()
    
    best_distance = df[' fittest_distance'].max()
    avg_distance = df[' mean_distance'].mean()
    
    best_vertical_distance = df[' fittest_vertical_distance'].max()
    avg_vertical_distance = df[' average_vertical_distance'].mean()
    
    avg_max_links = df[' max_links'].mean()
    avg_links = df[' average_links'].mean()
    
    # Create the summary table
    table = f"""
| Parameter | Value |
|-----------|-------|
| Iterations | {params['iterations']} |
| Population size | {params['population_size']} |
| Gene count | {params['gene_count']} |
| Simulation time | {params['simulation_time']} |
| Simulation | {params['simulation']} |
| Tournament size | {params['tournament_size']} |
| Point mutate | {params['point_mutate']} |
| Shrink mutate | {params['shrink_mutate']} |
| Grow mutate | {params['grow_mutate']} |
| Best Fitness | {best_fitness:.2f} |
| Average Fitness | {avg_fitness:.2f} |
| Best Mountain Fitness | {best_mountain_fitness:.2f} |
| Average Mountain Fitness | {avg_mountain_fitness:.2f} |
| Best Mountain Touching Fitness | {best_mountain_touching_fitness:.2f} |
| Average Mountain Touching Fitness | {avg_mountain_touching_fitness:.2f} |
| Best Size Fitness | {best_size_fitness:.2f} |
| Average Size Fitness | {avg_size_fitness:.2f} |
| Best Distance | {best_distance:.2f} |
| Average Distance | {avg_distance:.2f} |
| Best Vertical Distance | {best_vertical_distance:.2f} |
| Average Vertical Distance | {avg_vertical_distance:.2f} |
| Average Max Links | {avg_max_links:.2f} |
| Average Links | {avg_links:.2f} |
"""
    return table

# Parameters for each test
test_params = {
    'iterations': 10,
    'population_size': 75,
    'gene_count': 5,
    'simulation_time': 5000,
    'simulation': "The creature spawned just outside the mountain range with the Gaussian pyramid.",
    'tournament_size': 'pop_size * 0.1',
    'point_mutate': 0.2,
    'shrink_mutate': 0.15,
    'grow_mutate': 0.15
}

# Generate summary table
summary_table = create_summary_table(df5, test_params)

# Print the summary table
print(summary_table)


| Parameter | Value |
|-----------|-------|
| Iterations | 10 |
| Population size | 75 |
| Gene count | 5 |
| Simulation time | 5000 |
| Simulation | The creature spawned just outside the mountain range with the Gaussian pyramid. |
| Tournament size | pop_size * 0.1 |
| Point mutate | 0.2 |
| Shrink mutate | 0.15 |
| Grow mutate | 0.15 |
| Best Fitness | 80.27 |
| Average Fitness | 4.45 |
| Best Mountain Fitness | 35.06 |
| Average Mountain Fitness | 12.82 |
| Best Mountain Touching Fitness | 48.66 |
| Average Mountain Touching Fitness | 3.58 |
| Best Size Fitness | 6.29 |
| Average Size Fitness | 32.72 |
| Best Distance | 25.00 |
| Average Distance | 4.16 |
| Best Vertical Distance | 29.24 |
| Average Vertical Distance | -2.30 |
| Average Max Links | 181.93 |
| Average Links | 21.73 |



Basic test one

| Parameter | Value |
|-----------|-------|
| Iterations | 1000 |
| Population size | 10 |
| Gene count | 3 |
| Simulation time | 2400 |
| Simulation | The creature spawned just outside the mountain range with the Gaussian pyramid. |
| Tournament size | 3 |
| Point mutate | 0.1 |
| Shrink mutate | 0.25 |
| Grow mutate | 0.1 |
| Best Fitness | 81.09 |
| Average Fitness | 34.79 |
| Best Mountain Fitness | 38.40 |
| Average Mountain Fitness | 12.38 |
| Best Mountain Touching Fitness | 25.00 |
| Average Mountain Touching Fitness | 0.41 |
| Best Size Fitness | 25.00 |
| Average Size Fitness | 22.00 |
| Average Max Links | 9.18 |
| Average Links | 3.76 |

Basic Test two

| Parameter | Value |
|-----------|-------|
| Iterations | 150 |
| Population size | 50 |
| Gene count | 5 |
| Simulation time | 2400 |
| Simulation | The creature spawned just outside the mountain range with the Gaussian pyramid. |
| Tournament size | pop_size * 0.1 |
| Point mutate | 0.1 |
| Shrink mutate | 0.1 |
| Grow mutate | 0.1 |
| Best Fitness | 144.66 |
| Average Fitness | 33.97 |
| Best Mountain Fitness | 100.00 |
| Average Mountain Fitness | 12.96 |
| Best Mountain Touching Fitness | 25.00 |
| Average Mountain Touching Fitness | 3.28 |
| Best Size Fitness | 25.00 |
| Average Size Fitness | 17.73 |
| Average Max Links | 70.50 |
| Average Links | 14.92 |

Basic Test Three
| Parameter | Value |
|-----------|-------|
| Iterations | 30 |
| Population size | 100 |
| Gene count | 4 |
| Simulation time | 4800 |
| Simulation | The creature spawned just outside the mountain range with the Gaussian pyramid. |
| Tournament size | pop_size * 0.1 |
| Point mutate | 0.2 |
| Shrink mutate | 0.1 |
| Grow mutate | 0.1 |
| Best Fitness | 107.09 |
| Average Fitness | 34.01 |
| Best Mountain Fitness | 79.55 |
| Average Mountain Fitness | 12.77 |
| Best Mountain Touching Fitness | 25.00 |
| Average Mountain Touching Fitness | 5.48 |
| Best Size Fitness | 25.00 |
| Average Size Fitness | 15.76 |
| Average Max Links | 209.65 |
| Average Links | 38.06 |

Basic Test Four
| Parameter | Value |
|-----------|-------|
| Iterations | 1000 |
| Population size | 50 |
| Gene count | 4 |
| Simulation time | 4800 |
| Simulation | The creature spawned just outside the mountain range with the Gaussian pyramid. |
| Tournament size | pop_size * 0.1 |
| Point mutate | 0.2 |
| Shrink mutate | 0.1 |
| Grow mutate | 0.1 |
| Best Fitness | 60.49 |
| Average Fitness | 6.74 |
| Best Mountain Fitness | 56.91 |
| Average Mountain Fitness | 12.53 |
| Best Mountain Touching Fitness | 15.44 |
| Average Mountain Touching Fitness | 6.40 |
| Best Size Fitness | 15.44 |
| Average Size Fitness | 18.60 |
| Average Max Links | 11.86 |
| Average Links | 2.80 |

Basic test Five
| Parameter | Value |
|-----------|-------|
| Iterations | 10 |
| Population size | 75 |
| Gene count | 5 |
| Simulation time | 5000 |
| Simulation | The creature spawned just outside the mountain range with the Gaussian pyramid. |
| Tournament size | pop_size * 0.1 |
| Point mutate | 0.2 |
| Shrink mutate | 0.15 |
| Grow mutate | 0.15 |
| Best Fitness | 80.27 |
| Average Fitness | 4.45 |
| Best Mountain Fitness | 35.06 |
| Average Mountain Fitness | 12.82 |
| Best Mountain Touching Fitness | 48.66 |
| Average Mountain Touching Fitness | 3.58 |
| Best Size Fitness | 6.29 |
| Average Size Fitness | 32.72 |
| Best Distance | 25.00 |
| Average Distance | 4.16 |
| Best Vertical Distance | 29.24 |
| Average Vertical Distance | -2.30 |
| Average Max Links | 181.93 |
| Average Links | 21.73 |